src/geo.rs

```diff
@@ -0,0 +1,354 @@
+//! Defines types to use with the geospatial commands.
+
+use super::{ErrorKind, RedisResult};
+use types::{FromRedisValue, RedisWrite, ToRedisArgs, Value};
+
+macro_rules! invalid_type_error {
+    ($v:expr, $det:expr) => {{
+        fail!((
+            ErrorKind::TypeError,
+            "Response was of incompatible type",
+            format!("{:?} (response was {:?})", $det, $v)
+        ));
+    }};
+}
+
+/// Units used by [`geo_dist`][1] and [`geo_radius`][2].
+///
+/// [1]: ../trait.Commands.html#method.geo_dist
+/// [2]: ../trait.Commands.html#method.geo_radius
+pub enum Unit {
+    Meters,
+    Kilometers,
+    Miles,
+    Feet,
+}
+
+impl ToRedisArgs for Unit {
+    fn write_redis_args<W>(&self, out: &mut W)
+    where
+        W: ?Sized + RedisWrite,
+    {
+        let unit = match *self {
+            Unit::Meters => "m",
+            Unit::Kilometers => "km",
+            Unit::Miles => "mi",
+            Unit::Feet => "ft",
+        };
+        out.write_arg(unit.as_bytes());
+    }
+}
+
+/// A coordinate (longitude, latitude). Can be used with [`geo_pos`][1]
+/// to parse response from Redis.
+///
+/// [1]: ../trait.Commands.html#method.geo_pos
+///
+/// `T` is the type of the every value.
+///
+/// * You may want to use either `f64` or `f32` if you want to perform mathematical operations.
+/// * To keep the raw value from Redis, use `String`.
+#[derive(Debug, PartialEq)]
+pub struct Coord<T> {
+    pub longitude: T,
+    pub latitude: T,
+}
+
+impl<T> Coord<T> {
+    /// Create a new Coord with the (longitude, latitude)
+    pub fn lon_lat(longitude: T, latitude: T) -> Coord<T> {
```

Comment:
Hm, it's unfortunate that there's no way to restrict it to f64 or f32 without us including another crate or having another trait.

Reply:
However, we can ... maybe make the docs stronger implying this is useless with any other `T` and then we can always go the trait way sometime later.

Reply:
The [documentation of `Coord`](https://github.com/mitsuhiko/redis-rs/blob/3d2b2e053e894bd84689ef5cc26572e14b61c1c6/src/geo.rs#L49-L50) mentions it.

Using strings is also useful if you want to keep the raw value.

Reply:
👍